pkg/controller/workload/workload_processor.go

```diff
@@ -604,6 +711,13 @@ func (p *Processor) handleAddressTypeResponse(rsp *service_discovery_v3.DeltaDis
 		}
 	}
 
+	workloadInQueue := p.locality.GetFromWaitQueue() // locality LB mode
+	for workload := range workloadInQueue {
+		if err = p.handleWorkload(workload); err != nil {
+			log.Errorf("handle workload %s failed, err: %v", workload.ResourceName(), err)
```

Comment:
looks a little unefficient, you can do this iteration only when locality info has been set.

And can we only call `handleWorkloadNewBoundServices`


where do you clear workloads from queue?

Reply:
ok